operator/pkg/controllers/addons/addons.go

```diff
@@ -0,0 +1,100 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package addons
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/awslabs/kit/operator/pkg/apis/controlplane/v1alpha1"
+	"github.com/awslabs/kit/operator/pkg/controllers/master"
+	"github.com/awslabs/kit/operator/pkg/errors"
+	"github.com/awslabs/kit/operator/pkg/kubeprovider"
+	"github.com/awslabs/kit/operator/pkg/utils/keypairs"
+	"github.com/awslabs/kit/operator/pkg/utils/object"
+	"github.com/awslabs/kit/operator/pkg/utils/scheme"
+	"github.com/awslabs/kit/operator/pkg/utils/secrets"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Controller struct {
+	kubeClient *kubeprovider.Client
+	keypairs   *keypairs.Provider
+}
+
+func New(kubeClient *kubeprovider.Client) *Controller {
+	return &Controller{
+		kubeClient: kubeClient,
+		keypairs:   keypairs.Reconciler(kubeClient),
+	}
+}
+
+type reconciler func(context.Context, *Controller, *v1alpha1.ControlPlane) (err error)
+
+func (c *Controller) Reconcile(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
+	newClusterClient, err := c.createKubeClient(ctx, object.NamespacedName(
```

Comment:
can you not use `client.ObjectKey(object)`?

consider naming this variable kubeClient to simplify.